### PR TITLE
build 1.55.0 :snowflake:

### PR DESCRIPTION
**Pull request by boldorider4**
azureml-mlflow 1.55.0 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-3850]
- [Upstream repository](https://inspector.pypi.io/project/azureml-mlflow/1.54.0.post1/packages/7b/51/36789b81be0ae7429e3d6d280c0fcb1c1d9ce1aa8afd312344cdcd7974ff/azureml_mlflow-1.54.0.post1-py3-none-any.whl/)

### Explanation of changes:

- building straight from the `whl` archive
- missing GitHub project or equivalent
- the `METADATA` file needs to be patched because `jsonpickle` is a redundant dependency
- patched `METADATA` straight from the `build[.sh|.bat]`
- set license to non SPDX value MOSA (Microsoft Online Subscription Agreement) since no other SPDX value matches the license

[PKG-3850]: https://anaconda.atlassian.net/browse/PKG-3850?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{% set name = "azureml_mlflow" %}
```
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
{% set name = "azureml-mlflow" %}
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    - azure.storage.blob
+    - mlflow
+
+about:
+  home: https://pypi.org/project/azureml-mlflow
+  license: MOSA
+  license_family: Other
+  license_file: azureml_mlflow-1.55.0.dist-info/LICENSE.txt
+  summary: Azure Machine Learning SDK allows to build and run machine learning workflows with Azure Machine Learning
+           services.
```
Comment (lorepirri) [inline recipe/meta.yaml]:
```suggestion
  summary: Azure Machine Learning SDK allows to build and run machine learning workflows with Azure Machine Learning services.
```

File: recipe/0001-remove-unused-dependency.patch [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---	2024-02-23 19:59:42.113969609 +0100
++++	2024-02-23 19:59:44.738215960 +0100
+@@ -22,7 +22,6 @@
+ Requires-Python: >=3.7,<4.0
+ Description-Content-Type: text/x-rst
+ License-File: LICENSE.txt
+-Requires-Dist: jsonpickle
```
Comment (JeanChristopheMorinPerso) [inline recipe/0001-remove-unused-dependency.patch]: Why did you remove this? Considering this package is proprietery, I would leave things as is to avoid any unexpected breakage.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    - pip
+  commands:
+    - pip check
+  imports:
+    - azureml
+    - azureml.mlflow
+    - azure.core
+    - azure.mgmt.core
+    - azure.storage.blob
+    - mlflow
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:48]:
```suggestion
```
The only thing we should test for is the package itself.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+about:
+  home: https://pypi.org/project/azureml-mlflow
+  license: MOSA
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: `LicenseRef-Microsft-EULA` would probably be more accurate. `license` must be an SPDX supported license. (https://spdx.github.io/spdx-spec/v2-draft/SPDX-license-expressions/)

File: recipe/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    - python
+    - pip
+    - wheel
+  run:
+    - python
+    - six
+    - cryptography
+    - mlflow-skinny
+    - azure-identity
+    - azure-common
```
Comment (lorepirri) [inline recipe/meta.yaml]: I can see that `six` is needed indeed,

```suggestion
    - azure-common >=1.1,<2.0.0
```

according to 

```
Requires-Dist: jsonpickle
Requires-Dist: mlflow-skinny
Requires-Dist: azure-identity
Requires-Dist: msrest >=0.6.18
Requires-Dist: azure-core !=1.22.0,<2.0.0,>=1.8.0
Requires-Dist: azure-mgmt-core <2.0.0,>=1.2.0
Requires-Dist: azure-storage-blob <=12.19.0,>=12.5.0
Requires-Dist: azure-common <2.0.0,>=1.1
Requires-Dist: cryptography
Requires-Dist: python-dateutil <3.0.0,>=2.7.3
```
in https://inspector.pypi.io/project/azureml-mlflow/1.55.0/packages/6a/f8/8f405b06492672fbe30d6d16dc46ce59c0edc63b58037cf69dcfc6290b26/azureml_mlflow-1.55.0-py3-none-any.whl/azureml_mlflow-1.55.0.dist-info/METADATA

**Review by lorepirri — APPROVED** [review]
LGTM, except pinning for `azure-common`, and the name of the package

File: recipe/meta.yaml (hunk 6 above)
Comment (boldorider4) [reply recipe/meta.yaml]: Thank you!

File: recipe/meta.yaml (hunk 4 above)
Comment (boldorider4) [reply recipe/meta.yaml:48]: since all these packages are dependencies for this one and this one is the "real" deliverable and none of those submodules are tested in their respective recipes, I thought it wouldn't hurt to test them here.

It wouldn't be the first time that some components ends up causing issues because of some missing dependency or because the dependency wasn't properly tested.

File: recipe/meta.yaml (hunk 5 above)
Comment (boldorider4) [reply recipe/meta.yaml]: Thank you!

File: recipe/meta.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: @boldorider4 The rule is to try to stick with the original name on PyPI. For example, see https://pypi.org/project/azureml-mlflow/. With pip you can install both `azureml-mlflow` and `azureml_mlflow` because the ecosystem normalizes names (see https://packaging.python.org/en/latest/specifications/name-normalization/). But we don't need to use underscores in conda.

Comment (boldorider4) [reply recipe/meta.yaml]: I think I can pull the old trick with the multi-output package so both names are resolved.

File: recipe/0001-remove-unused-dependency.patch (hunk 3 above)
Comment (boldorider4) [reply recipe/0001-remove-unused-dependency.patch]: Would it be ok in this case to leave jsonpickle out? Or do you feel strongly about re-including it? I'm trying to understand if this is a case of "both schools of thought are valid".

File: recipe/meta.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: We don't need to support both. Keep the original name(`azureml-mlflow`). Adding both just adds unnecessary maintenance.

File: recipe/0001-remove-unused-dependency.patch (hunk 3 above)
Comment (JeanChristopheMorinPerso) [reply recipe/0001-remove-unused-dependency.patch]: I feel strongly about this one. I'd prefer if we don't diverge from what upstream says because there is too much unknowns.

Comment (boldorider4) [reply recipe/0001-remove-unused-dependency.patch]: Ok. So be it.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Comment by lorepirri** [comment]
@boldorider4 some of the dependencies are not available for the `py312` variant
